### DISCUSSION TITLE
Trigger Renovate on-demand on all configured repos

### DISCUSSION THREAD

**Discussion by gnaponie** [renovatebot/renovate#28229] [Suggest an Idea]
### Tell us more.

Hello,
I run a self-hosted Renovate instance and I'd need Renovate to be triggered on-demand on all repos (or a set of repos). But I don't want to be the repo maintainer to trigger Renovate on a dependency (for example from the Dependency Dashboard), but I would like Renovate to be triggered on-demand from an external service on all repos configured.

I'll be more clear on the use case: imagine I have a security bug that needs to be fixed immediately. As soon as the fix is ready I want $external_service to trigger Renovate (somehow) immediately for that fix only, ideally on all repos that include that bugged dependency. Or even on all the configured repos (which might not be optimized, but still ok).

I imagine to run Renovate (cron job) every 2 hours, but it is instructed to file PRs twice per week.

Is there a way to do that at the moment? Would it be complicated to add this kind of feature? Any suggestion is welcome.
Thank you
Cheers

**Comment by rarkins** [comment]
Sorry but I don't understand what you're asking. Assuming you mean the Renovate CLI, it will already run over every accessible repository each time you run it. i.e. "on all repos" as you're asking. What is it that's not working?

**Comment by gnaponie** [comment]
Thank you for your reply. I'll try to rephrase. I hope I didn't misunderstand the behavior, forgive me otherwise. The documentation explains the schedule is meant on a "global sense" (example a cron job set up by who owns the self-hosted renovate) and "specific sense" (when it will file PRs).
Imagine you have Renovate running every 2 hours (the cron job), but it's instructed to file PRs every Monday morning.
Is there a way I can say for all configured repos (even better only on specific ones, but all repos is fine too), just once, to file PRs right away, for the next run (and don't wait for Monday?).
I know you could ask in the repo itself (by the dependency dashboard for example), but can I (the owner of the self hosted renovate) be the one enforcing the PR to be filed on the next run (and not to wait on Monday) for a specific set or repos (or all configured repos)?

I hope it's clearer now.

**Reply by rarkins** [reply]
Everything I know about Renovate scheduling is in https://docs.renovatebot.com/key-concepts/scheduling/

If you want to force override configured schedules, you can use the "force" object

**Reply by viceice** [reply]
you can from bot config

https://docs.renovatebot.com/self-hosted-configuration/#checkedbranches

**Reply by gnaponie** [reply]
I see. I think `checkedBranches` it's exactly what I need! Thank you!
I'm just worried that this feature is flagged as experimental. Do you know if there's any future plan for it?
Thank you.

**Reply by rarkins** [reply]
We will probably graduate this to non-experimental soon as we're using it in our Mend hosted app.

**Reply by gnaponie** [reply]
That's terrific. Thank you. I think we can close this.
